internal/support/cronscheduler/application/schedulejob.go

```diff
@@ -183,3 +183,50 @@ func DeleteScheduleJobByName(ctx context.Context, name string, dic *di.Container
 	lc.Debugf("Successfully deleted the scheduled job: %s. Correlation-ID: %s", name, correlationId)
 	return nil
 }
+
+// LoadScheduleJobsToSchedulerManager loads all the existing schedule jobs to the scheduler manager
+func LoadScheduleJobsToSchedulerManager(ctx context.Context, dic *di.Container) errors.EdgeX {
+	dbClient := container.DBClientFrom(dic.Get)
+	schedulerManager := container.SchedulerManagerFrom(dic.Get)
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	ctx, correlationId := correlation.FromContextOrNew(ctx)
+
+	// TODO: issue-4834-6th - Decide the limit value should be a hard-coded value or the MaxResultCount value from configuration or support the sql statement that does not limit the number of rows
+	jobs, err := dbClient.AllScheduleJobs(context.Background(), 0, 999)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindDatabaseError, "failed to load all existing scheduled jobs", err)
+	}
+
+	// TODO: issue-4834-6th - Decide the limit value should be a hard-coded value or support the sql statement that does not limit the number of rows
+	allLatestRecords, err := dbClient.LatestScheduleActionRecords(ctx, 0, 999)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindDatabaseError, "failed to load latest schedule action records", err)
+	}
```

Comment:
move the allLatestRecords into the `if job.AdminState == models.Unlocked {`
change the API to `getLatestRecordsByJobName(ctx, jobName)`, and you don't need to set any limit to this function

If the job is Locked or before the startTime or after the endTime, we don't need to query the latestRecords